modules/component/spec/core/render-scheduler.spec.ts

```diff
@@ -1,13 +1,40 @@
-import { createRenderScheduler } from '../../src/core/render-scheduler';
+import {
+  createRenderScheduler,
+  RenderScheduler,
+} from '../../src/core/render-scheduler';
 import { NoopTickScheduler } from '../../src/core/tick-scheduler';
 import { MockChangeDetectorRef } from '../fixtures/fixtures';
+import { TestBed } from '@angular/core/testing';
+import { ChangeDetectorRef, Injectable } from '@angular/core';
 
 describe('createRenderScheduler', () => {
+  it('should initialize within injection context', () => {
+    @Injectable({ providedIn: 'root' })
+    class Service {
+      readonly renderScheduler = createRenderScheduler();
+    }
+
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: ChangeDetectorRef, useClass: MockChangeDetectorRef },
+      ],
+    });
+
+    const renderScheduler = TestBed.inject(Service).renderScheduler;
+    expect(renderScheduler instanceof RenderScheduler).toBe(true);
```

Comment:
This should also work:

```suggestion
    expect(renderScheduler).toBeInstanceOf(RenderScheduler);
```